homeassistant/components/thread/dataset_store.py

```diff
@@ -0,0 +1,150 @@
+"""Persistently store thread datasets."""
+from __future__ import annotations
+
+import dataclasses
+from datetime import datetime
+from typing import Any, cast
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.storage import Store
+from homeassistant.util import dt as dt_util, uuid as uuid_util
+
+from . import tlv_parser
+from .const import DOMAIN
+
+DATA_STORE = "datasets"
+STORAGE_KEY = "thread.datasets"
+STORAGE_VERSION_MAJOR = 1
+STORAGE_VERSION_MINOR = 1
+SAVE_DELAY = 10
+
+
+@dataclasses.dataclass(frozen=True)
+class DatasetEntry:
+    """Dataset store entry."""
+
+    preferred: bool
+    source: str
+    tlv: str
+
+    created: datetime = dataclasses.field(default_factory=dt_util.utcnow)
+    id: str = dataclasses.field(default_factory=uuid_util.random_uuid_hex)
+
+    @property
+    def extended_pan_id(self) -> str | None:
+        """Return extended PAN ID as a hex string."""
+        dataset = tlv_parser.parse_tlv(self.tlv)
```

Comment:
Can you add `parsed_tlv` as a `cached_property` to this class, so we don't parse it for each property?